scanomatic/ui_server_data/js/som/qc_normAPIHelper.js

```diff
@@ -1,9 +1,10 @@
 import $ from 'jquery';
-import * as d3 from 'd3';
 
 import { getLastSegmentOfPath } from './qc_normHelper';
 import { setSharedValue } from './helpers';
 
+const d3 = require('d3/d3.js');
```

Comment:
What is the issue? is it that we assign things to `d3.scanomatic`? In that case, perhaps use your new get/set shared values function instead?

Reply:
I'd really rather not have us use require

Reply:
That is one of the problems, but the crash happens because the code cannot address `d3.event` when in D3 is imported using modern style import. I think that is unrelated, and I can't see how using shared values for the scanomatic "plugin" would affect it. (Besides, the plugin is never accessed in the html, and the attributes set on it in one file are only accessed in that file, as far as I can tell.) That aside, doing so would also be a bit of a hacky solution, since it breaks with how you are (or were) supposed to make D3 plugins.

This stops being a problem in D3 v4, if I understand correctly, so I would say that this is transitional.

Reply:
so plan is to do oldschool require and then make task to migrate d3 to v4 and clean up the code.

Reply:
To me that seems better than hacking around the issue. There is already a task for this, to which a note could be added https://github.com/Scan-o-Matic/scanomatic-standalone/issues/6, or do you think this requires a more specific issue?

Reply:
just add a note